src/App.tsx

```diff
@@ -1,26 +1,165 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { USER_ID, postTodo, updateTodo } from './api/todos';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { useTodosContext } from './context/TodoContext';
+import { ErrorList } from './types/ErrorList';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const {
+    contextInputRef,
+    focusInput,
+    query,
+    setQuery,
+    todos,
+    setTodos,
+    errorMessage,
+    handleError,
+    setTempTodo,
+  } = useTodosContext();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+
+  useEffect(() => {
+    focusInput();
+  }, [isLoading]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleQuerySubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const trimedQuery = query.trim();
+
+    if (!trimedQuery.length) {
+      handleError(ErrorList.EmptyTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    const newItem = {
+      userId: USER_ID,
+      title: trimedQuery,
+      completed: false,
+    };
+
+    setTempTodo({ ...newItem, id: 0 });
+
+    postTodo({ ...newItem })
+      .then(todo => {
+        setTodos(prevTodos => [...prevTodos, todo]);
+        setQuery('');
+      })
+      .catch(() => {
+        handleError(ErrorList.AddTodo);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsLoading(false);
+      });
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleCheckAllFilter = (currentTodos: Todo[]) => {
```

Comment:
Let's improve the naming for this function. It's a bit unclear what this function does.

-----

Naming convention can be different from project to project but here are some basic rules:

`handle` prefix - used when handling DOM events like `onChange`, `onBlur`

`on` prefix - used for component props like `onTodoAdd`, `onTodoDelete`, `onError`

Other functions should start with some verbs like `get`, `format` and something like this